### PR TITLE
Expose factions to localisation

### PR DESCRIPTION
**Pull request by Mailaender**
plus a regression from https://github.com/OpenRA/OpenRA/pull/21221 fixed that appeared during string extraction and testing.

### CODE REVIEW

File: OpenRA.Game/Traits/World/Faction.cs [hunk 1]
```diff
@@ -29,6 +30,8 @@ public class FactionInfo : TraitInfo<Faction>
 		[Desc("The side that the faction belongs to. For example, England belongs to the 'Allies' side.")]
 		public readonly string Side = null;
 
+		[TranslationReference(optional: true)]
```
Comment (PunkPun) [inline OpenRA.Game/Traits/World/Faction.cs:33]: you made these optional but don't handle them being null. Is that something to be concerned about?

**Review by PunkPun — COMMENTED** [review]
untranslated in ingame menu

Screenshot 2024-07-17 at 19 06 13

File: OpenRA.Game/Traits/World/Faction.cs (hunk 1 above)
Comment (Mailaender) [reply OpenRA.Game/Traits/World/Faction.cs:33]: Yes, it hides a bug where the linter spits out
> OpenRA.Utility(1,1): Warning: Empty key in mod translation files required by Actor `world` trait `Faction.Description`
OpenRA.Utility(1,1): Warning: Empty key in mod translation files required by Actor `editorworld` trait `Faction.Description`

Comment (Mailaender) [reply OpenRA.Game/Traits/World/Faction.cs:33]: I added more null checks and made the `Name` not optional.

**Comment by Mailaender** [comment]
Fixed the `GameInfoStatsLogic`.

**Review by PunkPun — COMMENTED** [review]
I suggest following code usage to catch all places where factions are displayed. And I don't think we can translate immediately as there might be issues with multiplayer or replays

(replays menu)
Screenshot 2024-07-17 at 23 41 53

**Comment by Mailaender** [comment]
That dropdown is unavailable to me. I tried to localize it nonetheless.

**Comment by PunkPun** [comment]
you need to select player type apparently for it to become available

**Review by PunkPun — APPROVED** [review]

**Comment by PunkPun** [comment]
[changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/a15ece3c1f5241471d88ec2f3a480e384aea458f...879d057ac38186df5bddebc1409e9fffb8ce0a16)

**Pull request merged**